eda/tests/unittests/analysis/test_interaction.py

```diff
@@ -0,0 +1,198 @@
+    expected = AnalysisState(
+        {
+            "correlations": {
+                "train_data": pd.DataFrame(
+                    index=list("abcd"),
+                    data={
+                        "a": [1.00, 0.00, 0.77, -0.77],
+                        "b": [0.00, 1.00, 0.36, -0.36],
+                        "c": [0.77, 0.36, 1.00, -1.00],
+                        "d": [-0.77, -0.36, -1.00, 1.00],
+                    },
+                )
+            },
+            "correlations_method": "kendall",
+        }
+    )
+    __compare_outputs(expected, state)
+
+
+def test_Correlation_phik():
+    state = auto.analyze(train_data=__create_test_df(), return_state=True, anlz_facets=[Correlation(method="phik")])
+    state.correlations.train_data = state.correlations.train_data.round(2)
+    expected = AnalysisState(
+        {
+            "correlations": {
+                "train_data": pd.DataFrame(
+                    index=list("abcdef"),
+                    data={
+                        "a": [1.0, 1.0, 1.00, 1.00, 1.00, 1.00],
+                        "b": [1.0, 1.0, 0.00, 0.00, 0.00, 0.00],
+                        "c": [1.0, 0.0, 1.00, 0.79, 0.79, 0.79],
+                        "d": [1.0, 0.0, 0.79, 1.00, 0.79, 0.79],
+                        "e": [1.0, 0.0, 0.79, 0.79, 1.00, 0.79],
+                        "f": [1.0, 0.0, 0.79, 0.79, 0.79, 1.00],
+                    },
+                )
+            },
+            "correlations_method": "phik",
+        }
+    )
+    __compare_outputs(expected, state)
+
+
+def test_Correlation_focus():
+    actual = auto.analyze(
+        train_data=(__create_test_df()),
+        return_state=True,
+        anlz_facets=[Correlation(focus_field="c", focus_field_threshold=0.5)],
+    )
+    print(actual)
+    actual.correlations.train_data = actual.correlations.train_data.round(2)
+    actual.correlations_focus_high_corr.train_data = actual.correlations_focus_high_corr.train_data.round(2)
+    expected = AnalysisState(
+        {
+            "correlations": {
+                "train_data": pd.DataFrame(
+                    index=list("acd"),
+                    data={
+                        "a": [1.00, 0.88, -0.88],
+                        "c": [0.88, 1.00, -1.00],
+                        "d": [-0.88, -1.00, 1.00],
+                    },
+                )
+            },
+            "correlations_method": "spearman",
+            "correlations_focus_field": "c",
+            "correlations_focus_field_threshold": 0.5,
+            "correlations_focus_high_corr": {"train_data": pd.DataFrame(index=list("ad"), data={"c": [0.88, -1.00]})},
+        }
+    )
+    assert actual.correlations_focus_high_corr.train_data.equals(
+        expected.correlations_focus_high_corr.train_data
+    )  # noqa
+    actual.correlations_focus_high_corr.train_data = "--"
+    expected.correlations_focus_high_corr.train_data = "--"
+    __compare_outputs(expected, actual)
```

Comment:
Since this code is very similar to the `__compare_outputs()` function, you could consider making `__compare_outputs()` parameterized on fields that use `.equals()` and `==`.

Reply:
It is a good suggestion if it is reused many times, but in this specific case the extension is needed only for one case.